### PR TITLE
Handle pipe connecting error - Closes #620

### LINKED ISSUE #620: Command throws an error when connecting with pipe
### Description

When connecting the commands with pipe after the command, it will throws an error

`./bin/run transaction:sign '{"type":0,"timestamp":123,"senderPublicKey":"abcd1234","amount":"100","asset":{}}' | echo`

```
events.js:183
      throw er; // Unhandled 'error' event
      ^
Error: write EPIPE
    at WriteWrap.afterWrite [as oncomplete] (net.js:868:14)
```

**Reproduces how often:** [What percentage of the time does it reproduce?]
Always

### Versions

2.0-

### PR DESCRIPTION
**Pull request by mitsujutsu**
### What was the problem?

Node throws EPIPE error if process.stdout is being written to and a user runs it through a pipe that closes prematurely. 

### How did I fix it?

Gracefully handle EPIPE error.  Add extra check for stdout TTY mode.  

### Review checklist

* The PR resolves #620
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)

### CODE REVIEW

File: src/base.js [hunk 1]
```diff
@@ -35,6 +40,7 @@ export default class BaseCommand extends Command {
 		if (error) {
 			this.error(error.message ? error.message : error);
 		}
+		process.exit(0);
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/base.js]: ahh yea thats true... this one is not exactly related to `handleEPIPE`, sometime the process is not existing...Probably, need to look into the reason deeply too

File: src/commands/transaction/sign.js [hunk 2]
```diff
@@ -62,6 +57,11 @@ export default class SignCommand extends BaseCommand {
 					},
 		});
 
+		const transactionInput =
```
Comment (shuse2) [inline src/commands/transaction/sign.js]: Is it necessary to change the order?
If this is necessary, prefer to change order consistently other commands too.

Comment (willclarktech) [reply src/commands/transaction/sign.js]: @mitsujutsu In general we don't want to prompt a user for their sensitive inputs if we already know the other inputs are invalid.

File: test/utils/input/utils.js [hunk 3]
```diff
@@ -260,6 +253,18 @@ describe('input/utils utils', () => {
 				'Password was not successfully repeated.',
 			);
 		});
+
+		it('should reject with error when in TTY mode', () => {
+			utilHelpers.isTTY.restore();
```
Comment (willclarktech) [inline test/utils/input/utils.js]: I don't think this is actually necessary.

**Pull request merged**